app/main/views.py

```diff
@@ -11,13 +11,18 @@ def index():
 
 @main.route('/get-resources')
 def get_resources():
-    resources = Resource.query.all()
+    resources = Resource.query.all() 
     resources_as_dicts = Resource.get_resources_as_dicts(resources)
     return json.dumps(resources_as_dicts)
 
 
 @main.route('/get-associations/<int:resource_id>')
 def get_associations(resource_id):
-    # TODO: write logic to retrieve and return associations for resource.
-    associations = []
+    pin_info = Resource.query.filter_by(id=resource_id).first()
+    associations = {}
+    for pin in pin_info.text_descriptors:
```

Comment:
Before starting to iterate, add a check to make sure that a resource with such ID is found. Do it this way:
`if resource is None:
    return json.dumps(associations)`
i.e. if such resource doesn't exist, just return an empty object and exit.